core/dbt/task/runnable.py

```diff
@@ -406,11 +406,21 @@ def populate_adapter_cache(
         if not self.args.populate_cache:
             return
 
+        if self.manifest is None:
+            raise DbtInternalError("manifest was None in populate_adapter_cache")
+
         start_populate_cache = time.perf_counter()
+        # the cache only cares about executable nodes
+        cachable_nodes = [
+            node
+            for node in self.manifest.nodes.values()
+            if (node.is_relational and not node.is_ephemeral_model and not node.is_external_node)
+        ]
```

Comment:
any advantage to having this be a method or just a calculated attribute of the manifest itself?